lib/updatehub/device.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 O.S.Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __DEVICE_H__
+#define __DEVICE_H__
+
+#include <zephyr.h>
+#include <hwinfo.h>
+
+static inline bool updatehub_get_device_identity(char *id, int id_max_len)
+{
+	int i;
+	char buf[2];
+	int buf_len = 0;
+	u8_t hwinfo_id[id_max_len];
+	size_t length;
+
+
+	length = hwinfo_get_device_id(hwinfo_id, sizeof(hwinfo_id));
+	if (length <= 0) {
+		return false;
+	}
+
+	memset(id, 0, id_max_len);
+
+	for (i = 0; i < length; i++) {
+		snprintk(buf, id_max_len, "%02x", hwinfo_id[i]);
```

Comment:
Using id_max_len here would allow for an overrun of buf which is defined as `char buf[2];` above.  And if we were to change this to `sizeof(buf)` as it probably should be, then a size of 2 doesn't leave room for a terminating NULL.  Recommend making the definition:
`char buf[3];`
and here use
`snprintk(buf, sizeof(buf), "%02x", hwinfo_id[i]);`